simd/keccakf1600/f1600x4_amd64.go

```diff
@@ -30,7 +30,7 @@ type State struct {
 // Initialize the state and returns the buffer on which the four permutations
 // will act: a uint64 slice of length 100.  The first permutation will act
 // on {a[0], a[4], ..., a[96]}, the second on {a[1], a[5], ..., a[97]}, etc.
-func (s *State) Initialize() []uint64 {
```

Comment:
You can follow this pattern:
```
type X [100]uint64
func Initialize() X  { return X{} }
func (X) Permute() {}
```
and use it as:
```
X := keccakf1600.Initialize()
X[0] = 1
X.Permute()
```

Reply:
This doesn't work as `Initialize()` doesn't see the final address of the buffer it returns and so it cannot allign it properly.

Reply:
I was talking about the API pattern, not about the actual implementation.

Reply:
Yes, but any possible implementation of your API can't align the buffer correctly.